bundler/lib/bundler/spec_set.rb

```diff
@@ -40,6 +40,8 @@ def for(dependencies, check = false, match_current_platform = false)
         specs << spec
       end
 
+      specs.uniq! unless match_current_platform
```

Comment:
I was also considering a `Set` for `specs`, but I'm not familiar enough with the semantics. It also doesn't have methods I could see to add several items in one go.

Reply:
We used to use `Set`, but we removed it in order to support using the `set` gem inside `Gemfile`'s.